app/src/main/java/de/rampro/activitydiary/db/ActivityDiaryContentProvider.java

```diff
@@ -434,4 +441,77 @@ public int update(@NonNull Uri uri, @NonNull ContentValues values, @Nullable Str
+                String[] columnNames = allRowsStart.getColumnNames();
+                do {
+                    String id = null, start = null, end = null;
+                    for (String name : columnNames) {
+                        if (name.equals(ActivityDiaryContract.Diary.START))
+                            start = simpleDateFormat.format(allRowsStart.getLong(allRowsStart.getColumnIndex(name)));
+                        if (name.equals(ActivityDiaryContract.Diary.END))
+                            end = allRowsStart.getLong(allRowsStart.getColumnIndex(name)) == 0 ? "0" : simpleDateFormat.format(allRowsStart.getLong(allRowsStart.getColumnIndex(name)));
+                        if (name.equals(ActivityDiaryContract.Diary._ID))
+                            id = (allRowsStart.getString(allRowsStart.getColumnIndex(name)));
+                    }
+                    //System.out.println("id: " + id + ", start: " + start + ", end: " + end);
+
+                    //Values (date, start, end) are set to specific calendar - needed for 'between' searching
+                    // (ignore values that are equals to '0' -> if end date have '0' value it means that it has not set end date
+                    // what means that this activity is still in progress)
+                    try {
+                        if (date != null)
+                            searchedValueCal.setTime(simpleDateFormat.parse(date));
+                        if (!start.equals("0") && start != null)
+                            startValueCal.setTime(simpleDateFormat.parse(start));
+                        if (end != null && !end.equals("0"))
+                            endValueCal.setTime(simpleDateFormat.parse(end));
+                    }catch (ParseException e){
+                        e.printStackTrace();
+                    }
+
+                    //here it is creating query from IDs of those activities that are matching with searched date
+                    if ((searchedValueCal.after(startValueCal) && searchedValueCal.before(endValueCal) || searchedValueCal.equals(startValueCal) || searchedValueCal.equals(endValueCal))) {
+                        querySelection += querySelection.equals(" ") ?  ActivityDiaryContract.Diary.TABLE_NAME + "." +  ActivityDiaryContract.Diary._ID + "=" + id : " OR " + ActivityDiaryContract.Diary.TABLE_NAME + "." +  ActivityDiaryContract.Diary._ID + " =" + id ;
+                    } else if (searchedValueCal.after(startValueCal) && end.equals("0") && !searchedValueCal.equals("0")) {
+                        querySelection += querySelection.equals(" ") ?  ActivityDiaryContract.Diary.TABLE_NAME + "." +  ActivityDiaryContract.Diary._ID + " =" + id : "OR " + ActivityDiaryContract.Diary.TABLE_NAME + "." +  ActivityDiaryContract.Diary._ID + " =" + id ;
+                    }
+
+                } while (allRowsStart.moveToNext());
+            }
+        } catch (RuntimeException e) {
+            e.printStackTrace();
```

Comment:
this form of exception handling does not help the user... why catch this at all? If you can recover from it, please do so here. If not just let it throw...